src/script/api/script_company.cpp

```diff
@@ -308,11 +317,13 @@
 
 /* static */ bool ScriptCompany::SetPrimaryLiveryColour(LiveryScheme scheme, Colours colour)
 {
+	EnforcePrecondition(false, ScriptObject::GetCompany() != OWNER_DEITY);
 	return ScriptObject::Command<CMD_SET_COMPANY_COLOUR>::Do((::LiveryScheme)scheme, true, (::Colours)colour);
 }
 
 /* static */ bool ScriptCompany::SetSecondaryLiveryColour(LiveryScheme scheme, Colours colour)
 {
+	EnforcePrecondition(false, ScriptObject::GetCompany() != OWNER_DEITY);
```

Comment:
Maybe make a `ScriptObject::IsScriptCompanyScopeActive()` function? That would make the `EnforcePrecondition`s map better to the `@pre` comments. Though, since it already exists in other places in the code, it might not be wise to add that to this PR.